### DISCUSSION TITLE
Farmer says not synced or connected

### DISCUSSION THREAD

**Discussion by olivierLA75** [guydavis/machinaris#238] [Q&A] [answered]
My farmer on Windows 10 was synchronizing yesterday and finished over the night but it says not synced or connected to peers. The port forward on the router was done. The number of plots isn't updating. Also, I created a worker (Linux) from the UI and it doesn't show on the farmer/controller. Any idea of what I did wrong here?
Farmer/Controller hostname is SM1
<img width="1003" alt="Screen Shot 2021-08-20 at 9 29 44 AM" src="https://user-images.githubusercontent.com/83734065/130265391-ab2f81ab-be42-4f64-90a2-d9bc566e2cd2.png">
<img width="784" alt="Screen Shot 2021-08-20 at 9 29 58 AM" src="https://user-images.githubusercontent.com/83734065/130265399-95c16a1b-ca7a-4e2f-a737-50b387447fd2.png">
<img width="863" alt="Screen Shot 2021-08-20 at 9 30 19 AM" src="https://user-images.githubusercontent.com/83734065/130265413-462d4dfe-675a-4d7f-a5c0-d8438a866d14.png">
<img width="582" alt="Screen Shot 2021-08-20 at 9 34 30 AM" src="https://user-images.githubusercontent.com/83734065/130265554-b3ca1cf6-baca-4cfa-9d1c-8945280fdd2d.png">

**Comment by olivierLA75** [comment]
Actually solved the connection worker with the controller/farmer by adding the IP address in the docker run and it works. I can see the worker in the controller and it says that the worker is harvesting but the plots didn't add up in the summary. also, controller is showing a bigger number of plots in the farming section for the farmer that it's showing on the summary.1683 instead of 2395 (these are the plots on the farmer only) Farmer still not synced or connected to peers.
<img width="1019" alt="Screen Shot 2021-08-20 at 11 00 08 AM" src="https://user-images.githubusercontent.com/83734065/130274937-3bcac46c-143a-47a2-a17a-7ea98a8ba5b6.png">
<img width="1026" alt="Screen Shot 2021-08-20 at 11 00 20 AM" src="https://user-images.githubusercontent.com/83734065/130274999-131360f2-1663-46ef-ab96-d6583d4c6e72.png">
<img width="853" alt="Screen Shot 2021-08-20 at 10 59 58 AM" src="https://user-images.githubusercontent.com/83734065/130275023-956ea497-a382-48a3-97c5-3814cdb3a6df.png">

**Reply by guydavis** [reply]
Good day again.  Share you exact WS1 Docker launch command and also your WS1 ~/.machinaris/machinaris/logs/apisrv.log file.

Also, did you know you can simply click Reply in Github Disucssions?  You are starting a brand-new thread (Answer) over and over within every single Question you post.  So basically, you ask a Question (at very top), then every time you want to type something, you are entering an Answer, but you are not actually answering your own question.  So, I recommend you stop typing into this Answer box unless you are actually posting an Answer.

![image](https://user-images.githubusercontent.com/12936/130275818-c1cf6f29-2aec-4755-b853-ea3f04f7f6fb.png)

**Reply by olivierLA75** [reply]
got it for the reply/answer, will be careful about it in the future

Below is the WS1 launch command but not sure where to find the log file you're mentioning about. anyway I can see the worker on the farmers interface now, but not sure why the farmer says it's not synced or connected to peers

WS1 launch command:
root@ws1:/home/ws1# docker run \
>     -d -t --name machinaris \
>     -h WS1 \
>     --dns 64.71.0.34 \
>     -v ~/.machinaris:/root/.chia \
>     -v "/media/ws1/temp:/plotting" \
>     -v "/media/ws1/plotss/plot:/plots" \
>     -v "/media/ws1/MiningSSD/plotssd:/plots2" \
>     -v "/media/ws1/plots1/plotsss:/plots3" \
>     -v "/media/ws1/plots2/plots:/plots4" \
>     -e TZ=America/Los_Angeles \
>     -e farmer_address=SM1 \
>     -e farmer_port=8447 \
>     -e flax_farmer_port=6888 \
>     -e farmer_pk=935466d0c25beb8d0b654b5583fc67cc030ab65e31f23d418f4a6deac0a89007406ac855f7d3dcf02e0150f9a02da66e \
>     -e pool_pk=8233f72412fb6920addd18eba4f7e526808f4546f4e3f428110b467e0fa004a3010f7d39d37c9e2b7e1bc7bbcab32bfd \
>     -e pool_contract_address=xch13lha0w29qur00l4ep653xml8rk39nnzxr36mgzglwlp3tgylh5dsaltylv \
>     -e mode=harvester,plotter \
>     -e controller_host=10.10.1.168 \
>     -e controller_api_port=8927 \
>     -e worker_address=10.10.1.163 \
>     -e plots_dir=/plots \
>     -e blockchains=chia,flax \
>     -p 8926:8926 -p 8927:8927 \
>     ghcr.io/guydavis/machinaris
c548c0be80f5a82395abfbf948e47bf35131129f7c6119929400511d3a57cef3
root@ws1:/home/ws1#

**Reply by olivierLA75** [reply]
I was able to check the worker's plot (WS1) on the Farmer's (SM1) user interface and they got verified and counted but they don't appear in the summary of the farmer at all. the number of plots the farmer shows are the plots that were on it when I installed Machinaris yesterday and it started syncing, in the meantime I transferred more plots to the drives and they are showing in the farming area of SM1 but not on the summary either.

**Reply by guydavis** [reply]
Thanks. I've got a few hours before I go offline for the weekend.  What does `chia farm summary` show when executed in-container on the SM1 system?

**Reply by olivierLA75** [reply]
![Image 8-20-21 at 12 47 PM](https://user-images.githubusercontent.com/83734065/130286255-7178e925-c2c6-4dd9-991b-92ec1865e651.jpg)

**Reply by guydavis** [reply]
If I understand correctly, the Chia binary is responding with only local plots.  This tells me that the WS1 (worker) has not been correctly configured to harvest its plots over to the farmer.

First go the Workers page and click New Worker.  Just once to show the window. 
Then, try the following to start your worker.  
```
docker run \
-d -t --name machinaris \
-h WS1 \
-v ~/.machinaris:/root/.chia \
-v "/media/ws1/temp:/plotting" \
-v "/media/ws1/plotss/plot:/plots" \
-v "/media/ws1/MiningSSD/plotssd:/plots2" \
-v "/media/ws1/plots1/plotsss:/plots3" \
-v "/media/ws1/plots2/plots:/plots4" \
-e TZ=America/Los_Angeles \
-e farmer_address=SM1 \
-e farmer_port=8447 \
-e flax_farmer_port=6885 \
-e farmer_pk=935466d0c25beb8d0b654b5583fc67cc030ab65e31f23d418f4a6deac0a89007406ac855f7d3dcf02e0150f9a02da66e \
-e pool_pk=8233f72412fb6920addd18eba4f7e526808f4546f4e3f428110b467e0fa004a3010f7d39d37c9e2b7e1bc7bbcab32bfd \
-e pool_contract_address=xch13lha0w29qur00l4ep653xml8rk39nnzxr36mgzglwlp3tgylh5dsaltylv \
-e mode=harvester,plotter \
-e controller_host=10.10.1.168 \
-e controller_api_port=8927 \
-e worker_address=10.10.1.163 \
-e plots_dir=/plots \
-e blockchains=chia,flax \
-p 8926:8926 -p 8927:8927 \
ghcr.io/guydavis/machinaris
```
Then go to the Worker, then look at `docker logs machinaris` and also look at the ~/.machinaris/mainnet/log/debug.log.  You need to see successful connection of the harvester (worker/WS1) to the farmer (fullnode/controller/SM1).  Until that happens you're not farming the worker's remote plots.

**Reply by olivierLA75** [reply]
ok going to try this. the farmer isn't updating the plots that are in the farm either, and it stills say that it's either not synced or not connected to peers...

**Reply by olivierLA75** [reply]
oot@ws1:/home/ws1# docker logs machinaris
Configuring Chia...
Adding key at path: /root/.chia/mnemonic.txt
Added plot directory "/plots".
Did not find your farmer's certificates within /root/.chia/farmer_ca.
See: https://github.com/guydavis/machinaris/wiki/Workers#harvester
Farmer peer updated, make sure your harvester has the proper cert installed
Restart any running chia services for changes to take effect
uPnP disabled
Restart any running chia services for changes to take effect
Daemon not started yet
Starting daemon
chia_harvester: started
Configuring Flax...
Adding key at path: /root/.chia/mnemonic.txt
Traceback (most recent call last):
  File "/chia-blockchain/venv/bin/chia_harvester", line 33, in <module>
    sys.exit(load_entry_point('chia-blockchain', 'console_scripts', 'chia_harvester')())
  File "/chia-blockchain/chia/server/start_harvester.py", line 53, in main
    return run_service(**kwargs)
  File "/chia-blockchain/chia/server/start_service.py", line 255, in run_service
    return asyncio.run(async_run_service(*args, **kwargs))
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/chia-blockchain/chia/server/start_service.py", line 249, in async_run_service
    return await service.run()
  File "/chia-blockchain/chia/server/start_service.py", line 171, in run
    await self.start()
  File "/chia-blockchain/chia/server/start_service.py", line 145, in start
    self._reconnect_tasks = [
  File "/chia-blockchain/chia/server/start_service.py", line 146, in <listcomp>
    start_reconnect_task(self._server, _, self._log, self._auth_connect_peers) for _ in self._connect_peers
  File "/chia-blockchain/chia/server/reconnect_task.py", line 12, in start_reconnect_task
    peer_info = PeerInfo(socket.gethostbyname(peer_info_arg.host), peer_info_arg.port)
socket.gaierror: [Errno -2] Name or service not known
Added plot directory "/plots".
Did not find your farmer's certificates within /root/.flax/farmer_ca.
See: https://github.com/guydavis/machinaris/wiki/Workers#harvester
Farmer peer updated, make sure your harvester has the proper cert installed
Restart any running flax services for changes to take effect
uPnP disabled
Restart any running flax services for changes to take effect
Daemon not started yet
Starting daemon
flax_harvester: started
Configuring Plotman...
Configuring Chiadog...
Starting Chiadog...
Configuring Flaxdog...
Starting Flaxdog...
Traceback (most recent call last):
  File "/flax-blockchain/venv/bin/flax_harvester", line 33, in <module>
    sys.exit(load_entry_point('flax-blockchain', 'console_scripts', 'flax_harvester')())
  File "/flax-blockchain/flax/server/start_harvester.py", line 53, in main
    return run_service(**kwargs)
  File "/flax-blockchain/flax/server/start_service.py", line 255, in run_service
    return asyncio.run(async_run_service(*args, **kwargs))
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/flax-blockchain/flax/server/start_service.py", line 249, in async_run_service
    return await service.run()
  File "/flax-blockchain/flax/server/start_service.py", line 171, in run
    await self.start()
  File "/flax-blockchain/flax/server/start_service.py", line 145, in start
    self._reconnect_tasks = [
  File "/flax-blockchain/flax/server/start_service.py", line 146, in <listcomp>
    start_reconnect_task(self._server, _, self._log, self._auth_connect_peers) for _ in self._connect_peers
  File "/flax-blockchain/flax/server/reconnect_task.py", line 12, in start_reconnect_task
    peer_info = PeerInfo(socket.gethostbyname(peer_info_arg.host), peer_info_arg.port)
socket.gaierror: [Errno -2] Name or service not known
Starting Machinaris API server...
Starting Machinaris Web server...
Completed startup.  Browse to port 8926.
root@ws1:/home/ws1#

**Reply by guydavis** [reply]
Are you 100% sure that you clicked New Worker button on Workers page just before starting the Worker again?  Message above shows that certs from the Controller have not been downloaded to the Worker.  Hence connection is not made, so no farming of those plots is happening.

If this happens after trying again, you'll need to manually copy the certs from:
SM1: ~/.machinaris/mainnet/config/ssl/ca  -> copy this folder
copy the folder and rename it on WS1: 
WS1: ~/.machinaris/farmer_ca

Then restart WS1.

**Reply by olivierLA75** [reply]
yes, I liked on new worker and left the page opened and then launched the command on the other computer. will try again. any idea why the SM1 isn't farming and the local plot number isn't being updated?

**Reply by guydavis** [reply]
_any idea why the SM1 isn't farming and the local plot number isn't being updated?_

Why yes.  As I just wrote above already, the log above shows that certs from the Controller have not been downloaded to the Worker. Hence connection is not made, so no farming of those plots is happening.

If this happens after trying again, you'll need to manually copy the certs from:
SM1: ~/.machinaris/mainnet/config/ssl/ca -> copy this folder
copy the folder and rename it on WS1:
WS1: ~/.machinaris/farmer_ca

Then restart WS1.

**Reply by olivierLA75** [reply]
I just found out that it seems that the SM1 (Farmer) isn't online but it's connected to the internet and everything is working so not sure what's the issue, and maybe that's why it doesn't synchronize with the WS1
<img width="744" alt="Screen Shot 2021-08-20 at 2 23 19 PM" src="https://user-images.githubusercontent.com/83734065/130294838-480dbd4d-c106-475f-97ad-eb1973599d37.png">

**Comment by olivierLA75** [comment]
I was referring to the plots that are in the farmer/controller SM1 only, not the ones that are in the worker. It shows 1694 plots on the farm when there is more than 2000 without counting the ones from WS1

**Comment by olivierLA75** [comment]
I can’t find the .machinaris folder in the SM1 (Windows), do you know where it’s located by any chance?

**Reply by guydavis** [reply]
Thanks.  I'm sure you told me that SM1 was Windows days ago, however reminding helps.  Take a look at the Docker run command: 
`-v ${APPDATA}/.machinaris:/root/.chia `

So APPDATA on Windows is found by typing `%APPDATA%` into a Windows Explorer window.  Should show you into your user folder.

Alright, I am now offline for the weekend.  Best of luck.  Will check back Monday.

**Reply by olivierLA75** [reply]
Hello guy, sorry to bother you again. so everything is set up and the right number of plots (6848) is showing on the farmer/controller (SM1 under Windows) but it's only farming the plots that are on the farmer, as you can see below only the plots in the SM1 are passing the filer (2100 plots). Do you think the issue is the certificates? I look for what you told me but I don't have amachinaris folder on the SM1, only have a .chia folder that contains the certificates (see attached), is that the folder I need to copy? also, in my case only the farmer is on Windows and all the other machines are on Ubuntu. can I copy a folder from Windows and put it in Ubuntu? thanks in advance for your help :) 
<img width="1662" alt="Screen Shot 2021-08-25 at 12 10 03 PM" src="https://user-images.githubusercontent.com/83734065/130851131-5672ba2e-7dbf-4869-b5c5-066727730a38.png">
<img width="934" alt="Screen Shot 2021-08-25 at 12 06 21 PM" src="https://user-images.githubusercontent.com/83734065/130851144-1c0187e1-82f3-4ba5-94b7-d1bc8155b766.png">

**Reply by guydavis** [reply]
Hi, I will have an updated `:test` [version ](https://github.com/guydavis/machinaris/wiki/Releases) available by tomorrow which should improve logging around issues related to automated certificate copy from Machinaris fullnode to Machinaris harvester on start of the harvester container. 

Please ensure  you click the 'New Worker' button on the Workers page of your Machinaris fullnode, then quickly (minutes), restart the Machinaris container on your harvesters.  Look at the output of `docker logs machinaris` on the harvester to see if the certificates were downloaded, you'll see unzip output with filenames.

Basically, the folder above (ca) from Machinaris fullnode is automatically copied to ~/.machinaris/farmer_ca on the harvester and then imported on startup of the container. This used to be a manual copy, but has been automated and works in my testing and those of others.  Hope this helps.

**Reply by olivierLA75** [reply]
Hi Guy, how can I know when the test version is available? I looked everywhere but can find anything on Github

**Reply by guydavis** [reply]
Github shows[ test actions](https://github.com/guydavis/machinaris/actions/workflows/test.yaml), resulting in new images tagged `:test`. Today's was a couple of hours ago.  

That said, you really would be better served on the [Discord server](https://discord.gg/mX4AtMTt87) as it allows actual discussion.  Github is good for short Q&A as most.

**Reply by olivierLA75** [reply]
Even with the test version I can't get the harvesters to work, the plots are not being farmed :(
Can it be because the farmer is on Windows and all the other machines are on Linux?
When I do Docker log Machinaris it says few times to make sure that the harvester has the proper certificates installed and I saw this that seems to be an error:

File "/chia-blockchain/chia/server/reconnect_task.py", line 12, in start_reconnect_task
    peer_info = PeerInfo(socket.gethostbyname(peer_info_arg.host), peer_info_arg.port)
socket.gaierror: [Errno -2] Name or service not known

I feel stupid, I really can't figure this certificate issue...

**Reply by olivierLA75** [reply]
I'm trying to do the ca certificate folder thing manually but Windows/Docker denies access to it. Any idea how to do that?
<img width="789" alt="Screen Shot 2021-08-26 at 11 26 21 PM" src="https://user-images.githubusercontent.com/83734065/131082253-50c90f8d-a38d-443f-b844-c99f37eda594.png">

**Comment by olivierLA75** [comment]
On my farmer SM1, only Flax Blockchain is synced but not Chia, any way to resolve this?

**Reply by guydavis** [reply]
Is the Chia blockchain not syncing at all or just syncing slower than you would like?  Take a look at the Network page and check it every 10 mins or so.

If the former, then you can try a container restart.  If the latter, try to add a Chia node/introducer using the Add Connection field on Network | Connections page.

**Reply by olivierLA75** [reply]
Hey Guy, I was able to set up most of my system over the weekend but I still have a few questions:

1) I'm not sure if I set everything right. the Farmer/controller can see all the workers and the plots are now combined but I feel like only the plots of the famer (SM1) are farming. (see screenshot) Also, when checking the plots on the workers (all harvesters) from the controller interface I get some message error saying that the plots have a pool public key that isn't in the farmer's pk list. do you maybe have a screenshot of where and how I need to add all the keys in each worker settings?
2) the controller says in the alert tab that the harvesters appears to be offline, do you know what could cause that? when I check the workers tab it shows that they are all responding
![Screen Shot 2021-08-24 at 12 23 32 PM](https://user-images.githubusercontent.com/83734065/130677416-d3b96323-db98-43a8-b105-7c7dfafe4735.png)
![Screen Shot 2021-08-24 at 12 21 32 PM](https://user-images.githubusercontent.com/83734065/130677443-a25f9845-7b18-4553-ab4a-1234b5435042.png)
![Screen Shot 2021-08-24 at 12 16 24 PM](https://user-images.githubusercontent.com/83734065/130677455-70c653b1-610f-44ae-9edc-99679fbde678.png)

**Reply by guydavis** [reply]
Hi!  I've been busy improving Machinaris for an upcoming release.  In particular, have been adding additional logging and warnings to make clear some the situations you seem to be encountering.  Sorry for the trouble.

1. Assuming you are plotting for a pool and using [MadMax](https://github.com/guydavis/machinaris/wiki/MadMax#plotman-config), you want `pool_contract_address` to be uncommented and filled in for all workers using the Settings | Plotting page.
2. Harvester offline likely means not-farming, probably due to missing CA certificates on harvester copied from Machinaris fullnode.

**Comment by olivierLA75** [comment]
Perfect. Thank you!

**Comment by olivierLA75** [comment, marked as answer]
Noted. Going to try to find the Discord channel, thanks
